pkg/generate/app/cmd/newapp.go

```diff
@@ -1020,6 +915,9 @@ func (c *AppConfig) run(acceptors app.Acceptors) (*AppResult, error) {
 
 	pipelines, err := c.buildPipelines(imageRefs, env)
 	if err != nil {
+		if err == app.ErrNameRequired {
+			err = fmt.Errorf("can't suggest a valid name, please specify a name with --name")
```

Comment:
Because here we can include a mention to the CLI argument "--name", while keeping code in pkg/generate/app agnostic to the CLI. This comes from a [comment](https://github.com/openshift/origin/pull/5141#discussion_r42329538) from @smarterclayton in a similar situation.